FreeRTOS-Plus/Demo/FreeRTOS-IoT-Libraries-LTS-Beta2/mqtt/mqtt_multitask/DemoTasks/MultitaskMQTTExample.c

```diff
@@ -0,0 +1,1457 @@
+    }
+
+    return status;
+}
+
+/*-----------------------------------------------------------*/
+
+static bool topicFilterMatch( const char * pTopicName,
+                              uint16_t topicNameLength,
+                              const char * pTopicFilter,
+                              uint16_t topicFilterLength )
+{
+    bool status = false, matchFound = false;
+    uint16_t nameIndex = 0, filterIndex = 0;
+
+    while( ( nameIndex < topicNameLength ) && ( filterIndex < topicFilterLength ) )
+    {
+        /* Check if the character in the topic name matches the corresponding
+         * character in the topic filter string. */
+        if( pTopicName[ nameIndex ] == pTopicFilter[ filterIndex ] )
+        {
+            /* Handle special corner cases regarding wildcards at the end of
+             * topic filters, as documented by the MQTT protocol spec. */
+            matchFound = matchEndWildcards( pTopicFilter,
+                                            topicNameLength,
+                                            topicFilterLength,
+                                            nameIndex,
+                                            filterIndex,
+                                            &status );
+        }
+        else
+        {
+            /* Check for matching wildcards. */
+            matchFound = matchWildcards( pTopicFilter,
+                                         pTopicName,
+                                         topicNameLength,
+                                         filterIndex,
+                                         &nameIndex,
+                                         &status );
+        }
+
+        if( matchFound == true )
+        {
+            break;
+        }
+
+        /* Increment indexes. */
+        nameIndex++;
+        filterIndex++;
+    }
+
+    if( status == false )
+    {
+        /* If the end of both strings has been reached, they match. */
+        status = ( ( nameIndex == topicNameLength ) && ( filterIndex == topicFilterLength ) );
+    }
+
+    return status;
+}
+
+/*-----------------------------------------------------------*/
+
+static void prvSubscriptionManager( MQTTContext_t * pMqttContext,
+                                    MQTTPacketInfo_t * pPacketInfo,
+                                    uint16_t packetIdentifier,
+                                    MQTTPublishInfo_t * pPublishInfo )
+{
+    assert( pMqttContext != NULL );
+    assert( pPacketInfo != NULL );
+    AckInfo_t xAckInfo;
+    MQTTStatus_t xStatus = MQTTSuccess;
+    bool xIsMatched = false;
+    size_t i;
+    MQTTSubscribeInfo_t * pxSubscribeInfo = NULL;
+
+    /* Handle incoming publish. The lower 4 bits of the publish packet
+     * type is used for the dup, QoS, and retain flags. Hence masking
+     * out the lower bits to check if the packet is publish. */
+    if( ( pPacketInfo->type & 0xF0U ) == MQTT_PACKET_TYPE_PUBLISH )
+    {
+        assert( pPublishInfo != NULL );
```

Comment:
Regarding all the asserts in the demo, we use configAssert in the demos. So for consistency reasons, I think it will be a good change to convert all the asserts to configAssert

Reply:
Changed